testsuite/demo/run-pass/channel.verona

```diff
@@ -0,0 +1,156 @@
+/*****************************************************************************
+ * This example demonstrates how one could build a channel.
+ *
+ * It illustrates creating and using promises and cowns and communicating
+ * between behaviours.
+ *****************************************************************************/
+
+use "library/queue.verona"
+
+/*
+ * Create a channel from two queues:
+ *   - A queue of reads
```

Comment:
Are both queues required at the same time? Or could you play the same trick as in `Scheduler`?  It is fine to be different, just want to understand.

Reply:
You can play the same trick, I experimented with the single queue. This is, in some ways, just to be different but also the implementation with a single queue makes the read/write methods larger. The extra logic for changing the queue type distracts a little from the meaning of the read/write methods.